pkg/kapp/cmd/tools/inspect_tree_view.go

```diff
@@ -178,14 +178,18 @@ func (a *assocSortingValue) uidOwnersStr() string {
 			foundRes, found := a.rsByUID[string(ref.UID)]
 			if found {
 				// only nest into first object that we find
-				uids = append([]string{string(ref.UID)}, uids...)
+				identifiers = append([]string{a.resIdentifier(foundRes)}, identifiers...)
 				nextRes = &foundRes
 				break
 			}
 		}
 	}
 
-	return "ref-" + strings.Join(uids, "/")
+	return "ref-" + strings.Join(identifiers, "/")
+}
+
+func (a *assocSortingValue) resIdentifier(resource ctlres.Resource) string {
+	return fmt.Sprintf("%s-%s", resource.Kind(), resource.Name())
```

Comment:
Replacing UIDs with a `<Kind>-<Name>` value, this results in deterministic sorting. In the examples I considered, the resources with same name and kind in different namespaces had a different `kapp.k14s.io/association` label as well and the got grouped separately. We could certainly consider namespaces too.

```suggestion
return fmt.Sprintf("%s-%s-%s", resource.Namespace(), resource.Name(), resource.Kind())
```

Reply:
afaik it is currently using owner uid, which means that it's able to place related objects (aka owned) one after another (hence it forms a tree within a table). if we turn into ns/name/kind then it's not going to place related objects one after another and it's not going to be a tree. is this right, or am i missing something?

Reply:
We are calling this function for each owner resource. The function `uidOwnersStr()` returned a path of sorts using UIDs. Arranging one resource after the other.Replaced those UIDs with ns/name/kind  for each of the owners.
Instead of `UID1/UID2/UID3` it is now `ns-kind1-name1/ns-kind2-name2/ns-kind3-name3` (In case, this suggestion is accepted)

For example for a pod the entire group header value looks like:
`lbl-v1.0e236d647aa83adeae8c4c5bf1556f4d-2/child@ref-default-nginx-deployment-2-Deployment/default-nginx-deployment-2-cbdd67c76-ReplicaSet/default-nginx-deployment-2-cbdd67c76-k6pf4-Pod`

Maybe, we should consider stricter test cases for tree view?